pyop2/codegen/c/solve.c

```diff
@@ -0,0 +1,29 @@
+#include <petscsys.h>
+#include <petscblaslapack.h>
+#define BUF_SIZE 30
+
+static PetscBLASInt ipiv_buffer[BUF_SIZE];
+static PetscScalar work_buffer[BUF_SIZE*BUF_SIZE];
```

Comment:
If we include both solve and inverse, then these variables are declared twice, which I think will not work.

Similarly, the BUF_SIZE will be `#define`d twice. 

the `BUF_SIZE` is fine because you can `#undef` it at the end of the function.

For the work arrays we can either name them uniquely for these two files, or wrap them both in include guards. In the latter case we should also define BUF_SIZE only once (so that we don't accidentally change one but not the other).

```c
#ifndef PYOP2_WORK_ARRAYS
#define PYOP2_WORK_ARRAYS
#define BUF_SIZE 30
static PetscBLASInt ipiv_buffer[BUF_SIZE];
static PetscScalar work_buffer[BUF_SIZE*BUF_SIZE];
#endif
```